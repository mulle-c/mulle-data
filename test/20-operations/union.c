#include <mulle-data/mulle-data.h>

#include <stdio.h>

static void  mulle_range_print( struct mulle_range a)
{
   if( ! a.length)
      printf( "NULL");  
   else
      printf( "%td-%td (%td)", a.location, a.location + a.length - 1, a.length);  
}


int   main( int argc, char *argv[])
{
   struct mulle_range   a;
   struct mulle_range   b;
   struct mulle_range   c;
   unsigned int         i, n;

   for( a.location = 0; a.location < 100; a.location += 25)
      for( b.location = 0; b.location < 100; b.location += 25)
         for( a.length = 0; a.length < 100; a.length += 25)
            for( b.length = 0; b.length < 100; b.length += 25)
            {
               c = mulle_range_union( a, b);
               mulle_range_print( a);
               printf( " union ");
               mulle_range_print( b);
               printf( " = ");
               mulle_range_print( c);
               printf( "\n");
            }

   return( 0);
}


/*
 * extension : mulle-sde/c-test-library-demo
 * directory : demo/all
 * template  : .../hello.c
 * Suppress this comment with `export MULLE_SDE_GENERATE_FILE_COMMENTS=NO`
 */
