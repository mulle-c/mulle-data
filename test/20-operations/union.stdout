NULL union NULL = NULL
NULL union 0-24 (25) = 0-24 (25)
NULL union 0-49 (50) = 0-49 (50)
NULL union 0-74 (75) = 0-74 (75)
0-24 (25) union NULL = 0-24 (25)
0-24 (25) union 0-24 (25) = 0-24 (25)
0-24 (25) union 0-49 (50) = 0-49 (50)
0-24 (25) union 0-74 (75) = 0-74 (75)
0-49 (50) union NULL = 0-49 (50)
0-49 (50) union 0-24 (25) = 0-49 (50)
0-49 (50) union 0-49 (50) = 0-49 (50)
0-49 (50) union 0-74 (75) = 0-74 (75)
0-74 (75) union NULL = 0-74 (75)
0-74 (75) union 0-24 (25) = 0-74 (75)
0-74 (75) union 0-49 (50) = 0-74 (75)
0-74 (75) union 0-74 (75) = 0-74 (75)
NULL union NULL = NULL
NULL union 25-49 (25) = 25-49 (25)
NULL union 25-74 (50) = 25-74 (50)
NULL union 25-99 (75) = 25-99 (75)
0-24 (25) union NULL = 0-24 (25)
0-24 (25) union 25-49 (25) = 0-49 (50)
0-24 (25) union 25-74 (50) = 0-74 (75)
0-24 (25) union 25-99 (75) = 0-99 (100)
0-49 (50) union NULL = 0-49 (50)
0-49 (50) union 25-49 (25) = 0-49 (50)
0-49 (50) union 25-74 (50) = 0-74 (75)
0-49 (50) union 25-99 (75) = 0-99 (100)
0-74 (75) union NULL = 0-74 (75)
0-74 (75) union 25-49 (25) = 0-74 (75)
0-74 (75) union 25-74 (50) = 0-74 (75)
0-74 (75) union 25-99 (75) = 0-99 (100)
NULL union NULL = NULL
NULL union 50-74 (25) = 50-74 (25)
NULL union 50-99 (50) = 50-99 (50)
NULL union 50-124 (75) = 50-124 (75)
0-24 (25) union NULL = 0-24 (25)
0-24 (25) union 50-74 (25) = 0-74 (75)
0-24 (25) union 50-99 (50) = 0-99 (100)
0-24 (25) union 50-124 (75) = 0-124 (125)
0-49 (50) union NULL = 0-49 (50)
0-49 (50) union 50-74 (25) = 0-74 (75)
0-49 (50) union 50-99 (50) = 0-99 (100)
0-49 (50) union 50-124 (75) = 0-124 (125)
0-74 (75) union NULL = 0-74 (75)
0-74 (75) union 50-74 (25) = 0-74 (75)
0-74 (75) union 50-99 (50) = 0-99 (100)
0-74 (75) union 50-124 (75) = 0-124 (125)
NULL union NULL = NULL
NULL union 75-99 (25) = 75-99 (25)
NULL union 75-124 (50) = 75-124 (50)
NULL union 75-149 (75) = 75-149 (75)
0-24 (25) union NULL = 0-24 (25)
0-24 (25) union 75-99 (25) = 0-99 (100)
0-24 (25) union 75-124 (50) = 0-124 (125)
0-24 (25) union 75-149 (75) = 0-149 (150)
0-49 (50) union NULL = 0-49 (50)
0-49 (50) union 75-99 (25) = 0-99 (100)
0-49 (50) union 75-124 (50) = 0-124 (125)
0-49 (50) union 75-149 (75) = 0-149 (150)
0-74 (75) union NULL = 0-74 (75)
0-74 (75) union 75-99 (25) = 0-99 (100)
0-74 (75) union 75-124 (50) = 0-124 (125)
0-74 (75) union 75-149 (75) = 0-149 (150)
NULL union NULL = NULL
NULL union 0-24 (25) = 0-24 (25)
NULL union 0-49 (50) = 0-49 (50)
NULL union 0-74 (75) = 0-74 (75)
25-49 (25) union NULL = 25-49 (25)
25-49 (25) union 0-24 (25) = 0-49 (50)
25-49 (25) union 0-49 (50) = 0-49 (50)
25-49 (25) union 0-74 (75) = 0-74 (75)
25-74 (50) union NULL = 25-74 (50)
25-74 (50) union 0-24 (25) = 0-74 (75)
25-74 (50) union 0-49 (50) = 0-74 (75)
25-74 (50) union 0-74 (75) = 0-74 (75)
25-99 (75) union NULL = 25-99 (75)
25-99 (75) union 0-24 (25) = 0-99 (100)
25-99 (75) union 0-49 (50) = 0-99 (100)
25-99 (75) union 0-74 (75) = 0-99 (100)
NULL union NULL = NULL
NULL union 25-49 (25) = 25-49 (25)
NULL union 25-74 (50) = 25-74 (50)
NULL union 25-99 (75) = 25-99 (75)
25-49 (25) union NULL = 25-49 (25)
25-49 (25) union 25-49 (25) = 25-49 (25)
25-49 (25) union 25-74 (50) = 25-74 (50)
25-49 (25) union 25-99 (75) = 25-99 (75)
25-74 (50) union NULL = 25-74 (50)
25-74 (50) union 25-49 (25) = 25-74 (50)
25-74 (50) union 25-74 (50) = 25-74 (50)
25-74 (50) union 25-99 (75) = 25-99 (75)
25-99 (75) union NULL = 25-99 (75)
25-99 (75) union 25-49 (25) = 25-99 (75)
25-99 (75) union 25-74 (50) = 25-99 (75)
25-99 (75) union 25-99 (75) = 25-99 (75)
NULL union NULL = NULL
NULL union 50-74 (25) = 50-74 (25)
NULL union 50-99 (50) = 50-99 (50)
NULL union 50-124 (75) = 50-124 (75)
25-49 (25) union NULL = 25-49 (25)
25-49 (25) union 50-74 (25) = 25-74 (50)
25-49 (25) union 50-99 (50) = 25-99 (75)
25-49 (25) union 50-124 (75) = 25-124 (100)
25-74 (50) union NULL = 25-74 (50)
25-74 (50) union 50-74 (25) = 25-74 (50)
25-74 (50) union 50-99 (50) = 25-99 (75)
25-74 (50) union 50-124 (75) = 25-124 (100)
25-99 (75) union NULL = 25-99 (75)
25-99 (75) union 50-74 (25) = 25-99 (75)
25-99 (75) union 50-99 (50) = 25-99 (75)
25-99 (75) union 50-124 (75) = 25-124 (100)
NULL union NULL = NULL
NULL union 75-99 (25) = 75-99 (25)
NULL union 75-124 (50) = 75-124 (50)
NULL union 75-149 (75) = 75-149 (75)
25-49 (25) union NULL = 25-49 (25)
25-49 (25) union 75-99 (25) = 25-99 (75)
25-49 (25) union 75-124 (50) = 25-124 (100)
25-49 (25) union 75-149 (75) = 25-149 (125)
25-74 (50) union NULL = 25-74 (50)
25-74 (50) union 75-99 (25) = 25-99 (75)
25-74 (50) union 75-124 (50) = 25-124 (100)
25-74 (50) union 75-149 (75) = 25-149 (125)
25-99 (75) union NULL = 25-99 (75)
25-99 (75) union 75-99 (25) = 25-99 (75)
25-99 (75) union 75-124 (50) = 25-124 (100)
25-99 (75) union 75-149 (75) = 25-149 (125)
NULL union NULL = NULL
NULL union 0-24 (25) = 0-24 (25)
NULL union 0-49 (50) = 0-49 (50)
NULL union 0-74 (75) = 0-74 (75)
50-74 (25) union NULL = 50-74 (25)
50-74 (25) union 0-24 (25) = 0-74 (75)
50-74 (25) union 0-49 (50) = 0-74 (75)
50-74 (25) union 0-74 (75) = 0-74 (75)
50-99 (50) union NULL = 50-99 (50)
50-99 (50) union 0-24 (25) = 0-99 (100)
50-99 (50) union 0-49 (50) = 0-99 (100)
50-99 (50) union 0-74 (75) = 0-99 (100)
50-124 (75) union NULL = 50-124 (75)
50-124 (75) union 0-24 (25) = 0-124 (125)
50-124 (75) union 0-49 (50) = 0-124 (125)
50-124 (75) union 0-74 (75) = 0-124 (125)
NULL union NULL = NULL
NULL union 25-49 (25) = 25-49 (25)
NULL union 25-74 (50) = 25-74 (50)
NULL union 25-99 (75) = 25-99 (75)
50-74 (25) union NULL = 50-74 (25)
50-74 (25) union 25-49 (25) = 25-74 (50)
50-74 (25) union 25-74 (50) = 25-74 (50)
50-74 (25) union 25-99 (75) = 25-99 (75)
50-99 (50) union NULL = 50-99 (50)
50-99 (50) union 25-49 (25) = 25-99 (75)
50-99 (50) union 25-74 (50) = 25-99 (75)
50-99 (50) union 25-99 (75) = 25-99 (75)
50-124 (75) union NULL = 50-124 (75)
50-124 (75) union 25-49 (25) = 25-124 (100)
50-124 (75) union 25-74 (50) = 25-124 (100)
50-124 (75) union 25-99 (75) = 25-124 (100)
NULL union NULL = NULL
NULL union 50-74 (25) = 50-74 (25)
NULL union 50-99 (50) = 50-99 (50)
NULL union 50-124 (75) = 50-124 (75)
50-74 (25) union NULL = 50-74 (25)
50-74 (25) union 50-74 (25) = 50-74 (25)
50-74 (25) union 50-99 (50) = 50-99 (50)
50-74 (25) union 50-124 (75) = 50-124 (75)
50-99 (50) union NULL = 50-99 (50)
50-99 (50) union 50-74 (25) = 50-99 (50)
50-99 (50) union 50-99 (50) = 50-99 (50)
50-99 (50) union 50-124 (75) = 50-124 (75)
50-124 (75) union NULL = 50-124 (75)
50-124 (75) union 50-74 (25) = 50-124 (75)
50-124 (75) union 50-99 (50) = 50-124 (75)
50-124 (75) union 50-124 (75) = 50-124 (75)
NULL union NULL = NULL
NULL union 75-99 (25) = 75-99 (25)
NULL union 75-124 (50) = 75-124 (50)
NULL union 75-149 (75) = 75-149 (75)
50-74 (25) union NULL = 50-74 (25)
50-74 (25) union 75-99 (25) = 50-99 (50)
50-74 (25) union 75-124 (50) = 50-124 (75)
50-74 (25) union 75-149 (75) = 50-149 (100)
50-99 (50) union NULL = 50-99 (50)
50-99 (50) union 75-99 (25) = 50-99 (50)
50-99 (50) union 75-124 (50) = 50-124 (75)
50-99 (50) union 75-149 (75) = 50-149 (100)
50-124 (75) union NULL = 50-124 (75)
50-124 (75) union 75-99 (25) = 50-124 (75)
50-124 (75) union 75-124 (50) = 50-124 (75)
50-124 (75) union 75-149 (75) = 50-149 (100)
NULL union NULL = NULL
NULL union 0-24 (25) = 0-24 (25)
NULL union 0-49 (50) = 0-49 (50)
NULL union 0-74 (75) = 0-74 (75)
75-99 (25) union NULL = 75-99 (25)
75-99 (25) union 0-24 (25) = 0-99 (100)
75-99 (25) union 0-49 (50) = 0-99 (100)
75-99 (25) union 0-74 (75) = 0-99 (100)
75-124 (50) union NULL = 75-124 (50)
75-124 (50) union 0-24 (25) = 0-124 (125)
75-124 (50) union 0-49 (50) = 0-124 (125)
75-124 (50) union 0-74 (75) = 0-124 (125)
75-149 (75) union NULL = 75-149 (75)
75-149 (75) union 0-24 (25) = 0-149 (150)
75-149 (75) union 0-49 (50) = 0-149 (150)
75-149 (75) union 0-74 (75) = 0-149 (150)
NULL union NULL = NULL
NULL union 25-49 (25) = 25-49 (25)
NULL union 25-74 (50) = 25-74 (50)
NULL union 25-99 (75) = 25-99 (75)
75-99 (25) union NULL = 75-99 (25)
75-99 (25) union 25-49 (25) = 25-99 (75)
75-99 (25) union 25-74 (50) = 25-99 (75)
75-99 (25) union 25-99 (75) = 25-99 (75)
75-124 (50) union NULL = 75-124 (50)
75-124 (50) union 25-49 (25) = 25-124 (100)
75-124 (50) union 25-74 (50) = 25-124 (100)
75-124 (50) union 25-99 (75) = 25-124 (100)
75-149 (75) union NULL = 75-149 (75)
75-149 (75) union 25-49 (25) = 25-149 (125)
75-149 (75) union 25-74 (50) = 25-149 (125)
75-149 (75) union 25-99 (75) = 25-149 (125)
NULL union NULL = NULL
NULL union 50-74 (25) = 50-74 (25)
NULL union 50-99 (50) = 50-99 (50)
NULL union 50-124 (75) = 50-124 (75)
75-99 (25) union NULL = 75-99 (25)
75-99 (25) union 50-74 (25) = 50-99 (50)
75-99 (25) union 50-99 (50) = 50-99 (50)
75-99 (25) union 50-124 (75) = 50-124 (75)
75-124 (50) union NULL = 75-124 (50)
75-124 (50) union 50-74 (25) = 50-124 (75)
75-124 (50) union 50-99 (50) = 50-124 (75)
75-124 (50) union 50-124 (75) = 50-124 (75)
75-149 (75) union NULL = 75-149 (75)
75-149 (75) union 50-74 (25) = 50-149 (100)
75-149 (75) union 50-99 (50) = 50-149 (100)
75-149 (75) union 50-124 (75) = 50-149 (100)
NULL union NULL = NULL
NULL union 75-99 (25) = 75-99 (25)
NULL union 75-124 (50) = 75-124 (50)
NULL union 75-149 (75) = 75-149 (75)
75-99 (25) union NULL = 75-99 (25)
75-99 (25) union 75-99 (25) = 75-99 (25)
75-99 (25) union 75-124 (50) = 75-124 (50)
75-99 (25) union 75-149 (75) = 75-149 (75)
75-124 (50) union NULL = 75-124 (50)
75-124 (50) union 75-99 (25) = 75-124 (50)
75-124 (50) union 75-124 (50) = 75-124 (50)
75-124 (50) union 75-149 (75) = 75-149 (75)
75-149 (75) union NULL = 75-149 (75)
75-149 (75) union 75-99 (25) = 75-149 (75)
75-149 (75) union 75-124 (50) = 75-149 (75)
75-149 (75) union 75-149 (75) = 75-149 (75)
